src/libraries/System.Drawing.Common/src/System/Drawing/NativeMethods.cs

```diff
@@ -34,14 +34,13 @@ internal struct BITMAPINFO_FLAT
             public int bmiHeader_biClrUsed;
             public int bmiHeader_biClrImportant;
 
-            [MarshalAs(UnmanagedType.ByValArray, SizeConst = BITMAPINFO_MAX_COLORSIZE * 4)]
-            public byte[] bmiColors; // RGBQUAD structs... Blue-Green-Red-Reserved, repeat...
+            public fixed byte bmiColors[BITMAPINFO_MAX_COLORSIZE * 4]; // RGBQUAD structs... Blue-Green-Red-Reserved, repeat...
```

Comment:
It looks like we only ever consume it as a `ref` from some `DllImport`, but an arbitrary user inspecting this might think its "correct" or generally okay to follow.

But flattening structs isn't always safe (padding) nor is making dynamic length structs be larger by default (arrays).

Reply:
This is more performant equivalent of the existing code. It was not just used as `ref` from `DllImport` - it was also referenced here: https://github.com/dotnet/runtime/pull/33967/files#diff-418606aa6cb2e6e1de0fb92541ed8538L108

>  flattening structs isn't always safe (padding) nor is making dynamic length structs be larger by default (arrays).

I agree with this in general. I do not see a problem with how it it is used here.